runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -847,9 +847,9 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 		bool const compressed = compressObjectReferences();
 		void **dataAddrPtr = NULL;
 		if (compressed) {
-			dataAddrPtr = &((J9IndexableObjectContiguousCompressed *)arrayPtr)->dataAddr;
+			dataAddrPtr = &((J9IndexableObjectWithDataAddressContiguousCompressed *)arrayPtr)->dataAddr;
 		} else {
-			dataAddrPtr = &((J9IndexableObjectContiguousFull *)arrayPtr)->dataAddr;
+			dataAddrPtr = &((J9IndexableObjectWithDataAddressContiguousFull *)arrayPtr)->dataAddr;
```

Comment:
How can 32-bit builds succeed (where `J9IndexableObjectWithDataAddressContiguousFull` is not declared)?

Reply:
They are able to succeed because of the `#if defined(J9VM_ENV_DATA64)` on line `835`.